### ISSUE TITLE
[macOS] Flickering when closing tabs

### ISSUE THREAD

**Issue by jasonhan-vassar** [alacritty/alacritty#7270]
When closing alacritty tabs, there is sometimes flickering. I think this is a regression since the older alacritty version I'm using doesn't have this issue (or it's not as noticeable).

Is this a winit or alacritty issue? Is there any interest in me investigating further?

https://github.com/alacritty/alacritty/assets/34594853/927b67e8-f706-487f-8b5f-dbfaa347c5ce

### System

OS: macOS
Version: alacritty 0.13.0-dev (ace987f3)

**Comment by chrisduerr** [comment]
Looking at the frames for this issue, it seems like the issue is with the Alacritty instance being closed becoming transparent before the tab is removed:

![tmp](https://github.com/alacritty/alacritty/assets/8886672/02081151-ac64-40bf-9ae0-d8e059fa5161)

Did you recently update your macOS version? Can you try to bisect?

**Comment by jasonhan-vassar** [comment]
I'm on macOS 13.5 and I haven't updated in a little while.

I did `git bisect` and I found the faulty commit! It's 2377c0a7280f31ccc46953c462823ee3a1083af5 (Bump winit to 0.28.3). This is from all the way back in 3/17/23, so it's actually not that recent.

However, even the "good" commits can *rarely* flicker if you spam opening/closing tabs enough. Bisecting some more, it seems that 1c878a0476257d328113efcd7088f1aa53f42e56 is the start of this.

The commit that introduces the new window feature, d5b2dac73a57ac72c1e9587ee5b53f9ca9f4a6e3, is perfect though. There is absolutely zero flickering.

It turns out I didn't notice until now because my fork uses `.with_transparent(false)` in `display/window.rs`, which makes the flickering opaque and very hard to notice.

This seems like a winit issue. I can try looking into it more when I have the time!

**Comment by kchibisov** [comment]
@jasonhan-vassar could you also try https://github.com/alacritty/alacritty/pull/7220 ?

**Comment by jasonhan-vassar** [comment]
Sure thing. I tried it and it still seems to flicker:

https://github.com/alacritty/alacritty/assets/34594853/74f2cfcc-549e-4599-9914-4f5ec7e2f8bd

**Comment by mrded** [comment]
Hold on. How do you open tabs? :) 
Seems like I missed something. I thought Alacritty will never have tabs.

**Comment by chrisduerr** [comment]
Alacritty will never have built-in tabs. But macOS supports tiling natively in the OS and we leverage this.

**Comment by mrded** [comment]
> macOS supports tiling natively in the OS and we leverage this.

How can I use it?

**Comment by chrisduerr** [comment]
You need to use the `SpawnNewWindow` binding or IPC command to create new windows rather than instances and configure macOS to group multiple windows of the same application as tabs.

**Comment by mrded** [comment]
Thank you! here is how to add tabs to anything:
https://support.apple.com/en-gb/guide/mac-help/mchla4695cce/mac

TL;DR: `Prefer tabs when opening documents: Always`

**Comment by cartesius68** [comment]
same issue

```
ProductName:		macOS
ProductVersion:		14.3.1
BuildVersion:		23D60
```

```
alacritty 0.14.0-dev (cb3a79db)
```

**Comment by nbuesing** [comment]
new to Alacritty, love it. 

Seeing this flickering as well

0.13.2
Sanoma (14.4.1)
Mac Pro M1 Silicon

**Comment by nixpulvis** [comment]
While #7928 doesn't fix this issue, it does reduce its noticeability if you're not using a transparent window.
